crates/bevy_reflect/src/serde/de/processor.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{PartialReflect, TypeRegistration, TypeRegistry};
+
+/// Allows overriding the default deserialization behaviour of
+/// [`ReflectDeserializer`] and [`TypedReflectDeserializer`] for specific
+/// [`TypeRegistration`]s.
+///
+/// When deserializing a reflected value, you may want to override the default
+/// behaviour and use your own logic for deserialization. This logic may also
+/// be context-dependent, and only apply for a single use of your
+/// [`ReflectDeserializer`]. To achieve this, you can create a processor and
+/// pass it in to your deserializer.
+///
+/// Whenever the deserializer attempts to deserialize a value, it will first
+/// call [`try_deserialize`] on your processor, which may take ownership of the
+/// deserializer and give back a [`Box<dyn PartialReflect>`], or return
+/// ownership of the deserializer back, and continue with the default logic.
+///
+/// # Compared to [`DeserializeWithRegistry`]
+///
+/// [`DeserializeWithRegistry`] allows you to define how your type will be
+/// deserialized by a [`TypedReflectDeserializer`], given the extra context of
+/// the [`TypeRegistry`]. If your type can be deserialized entirely from that,
+/// then you should prefer implementing that trait instead of using a processor.
+///
+/// However, you may need more context-dependent data which is only present in
+/// the scope where you create the [`TypedReflectDeserializer`]. For example, in
+/// an asset loader, the `&mut LoadContext` you get is only valid from within
+/// the `load` function. This is where a processor is useful, as the processor
+/// can capture local variables.
```

Comment:
Another difference that may or may not be worth mentioning is overriding behavior.

For example, I think with this PR you could override the custom `ReflectDeserialize` or `ReflectDeserializeWithRegistry` implementations with a custom processor.